applications/IgaApplication/custom_conditions/support_penalty_condition.cpp

```diff
@@ -0,0 +1,176 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ricky Aristio
+//                   Tobias Teschemacher
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_conditions/support_penalty_condition.h"
+
+namespace Kratos
+{
+    void SupportPenaltyCondition::CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag
+    )
+    {
+        KRATOS_TRY
+        const double penalty = GetProperties()[PENALTY_FACTOR];
+
+        const auto& r_geometry = GetGeometry();
+        const SizeType number_of_nodes = r_geometry.size();

```

Comment:
why not use the `PointsNumber` here also?

Reply:
not that this makes a lot of sense, but with number_of_nodes we mean the nodes/control points. With PointNumber we typically refer to the integration points. I tried to adapt this a bit from the structural mechanics application.